pulumi-deck/src/PulumiStageForm.tsx

```diff
@@ -0,0 +1,214 @@
+  }
+
+  private handleAccountChange(selectedAccount: string) {
+    this.props.formik.setFieldValue('parameters.account', selectedAccount);
+    const details = (this.state.accounts || []).find((account) => account.name === selectedAccount);
+    if (!details) {
+      return;
+    }
+
+    const namespaces = (details.namespaces || []).sort();
+    const namespace = (this.props.formik.values['parameters'] as IPulumiPluginProps).namespace;
+    if (!this.isExpression(namespace) && namespaces.every((ns) => ns !== namespace)) {
+      this.props.formik.setFieldValue('parameters.namespace', null);
+    }
+
+    this.setStateAndUpdateStage({
+      namespaces,
+    });
+  }
+
+  private handleNamespaceChange(selectedNamespace: string) {
+    this.props.formik.setFieldValue('parameters.namespace', selectedNamespace);
+  }
+
+  public render() {
+    const { account, namespace } = this.props.formik.values['parameters'];
+    const { accounts, namespaces } = this.state;
+
+    return (
+      <>
+        <h4>Account and Namespace</h4>
+        <StageConfigField label="Account">
+          <AccountSelectInput
+            renderFilterableSelectThreshold={1}
+            value={account}
+            onChange={(evt: any) => this.handleAccountChange(evt.target.value)}
+            accounts={accounts}
+            provider="'kubernetes'"
+          />
+        </StageConfigField>
+        <StageConfigField label="Namespace">
+          <ReactSelectInput
+            clearable={false}
+            value={namespace}
+            options={namespaces.map((ns) => ({ value: ns, label: ns }))}
+            onChange={(evt: any) => this.handleNamespaceChange(evt.target.value)}
+          />
+        </StageConfigField>
+
+        <hr />
+
+        <h4>Repository Options</h4>
+        <FormikFormField
+          help={
+            <HelpField content="The working directory where the repository will be cloned into. All commands will be run in the context of this directory." />
+          }
+          input={(props) => <TextInput {...props} />}
+          label="Working Directory"
+          name="parameters.workingDir"
+        />
+        <FormikFormField
+          help={
+            <HelpField
+              content="The repo URL to clone from. For HTTPS URLs, specify the username and password as placeholders in the URL 
+            that map to the key names of secrets (see the Secrets Name parameter below). For SSH URLs, the plugin configuration must be
+            updated to specify the volume name for the SSH configuration files to use. The volume would be mounted as ~/.ssh in the
+            container."
+            />
+          }
+          input={(props) => <TextInput {...props} />}
+          label="Git Repo URL"
+          name="parameters.gitRepoUrl"
+        />
+        <FormikFormField
+          help={<HelpField content="The command to execute in order to restore dependencies for the Pulumi app." />}
```

Comment:
I wonder if we should include an example like 'npm ci' or something since it might not be obvious what this means? On the other hand, I don't really want to have too many examples.